tf5muxserver/mux_server_GetResourceIdentitySchemas.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package tf5muxserver
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-go/tfprotov5"
+
+	"github.com/hashicorp/terraform-plugin-mux/internal/logging"
+)
+
+// GetResourceIdentitySchemas merges the schemas returned by the
+// tfprotov5.ResourceIdentitySchema associated with muxServer into a single schema.
+// Everything must be returned from only one server.
+// Schemas must be identical between all servers.
+func (s *muxServer) GetResourceIdentitySchemas(ctx context.Context, req *tfprotov5.GetResourceIdentitySchemasRequest) (*tfprotov5.GetResourceIdentitySchemasResponse, error) {
+	rpc := "GetResourceIdentitySchemas"
+	ctx = logging.InitContext(ctx)
+	ctx = logging.RpcContext(ctx, rpc)
+
+	s.serverDiscoveryMutex.Lock()
+	defer s.serverDiscoveryMutex.Unlock()
+
+	resp := &tfprotov5.GetResourceIdentitySchemasResponse{
+		IdentitySchemas: map[string]*tfprotov5.ResourceIdentitySchema{},
+		Diagnostics:     []*tfprotov5.Diagnostic{},
+	}
+
+	for _, server := range s.servers {
+		ctx := logging.Tfprotov5ProviderServerContext(ctx, server)
+		logging.MuxTrace(ctx, "calling downstream server")
+
+		// TODO: Remove and call server.GetResourceIdentitySchemas below directly once interface becomes required.
+		//nolint:staticcheck // Intentionally verifying interface implementation
+		resourceIdentityServer, err := server.(tfprotov5.ProviderServerWithResourceIdentity).GetResourceIdentitySchemas(ctx, req)
```

Comment:
I would probably opt to check this interface and safely return a diagnostic if `server` doesn't implement `tfprotov5.ProviderServerWithResourceIdentity`, otherwise you'll get a panic here (similar to how `UpgradeResourceIdentity` is implemented)

Reply:
added a commit, please have a look if that's what you had in mind 😇

Reply:
Yep! 4c31d1fdf133ceee4a0f36fb540bc10311a8f334 was exactly what I meant 👍🏻